src/Workspaces/CSharp/Portable/Extensions/BlockSyntaxExtensions.cs

```diff
@@ -45,11 +43,29 @@ internal static class BlockSyntaxExtensions
                 }
             }
 
-            arrowExpression = null;
+            expression = null;
             semicolonToken = default;
             return false;
         }
 
+        public static bool TryConvertToArrowExpressionBody(
+            this BlockSyntax block, SyntaxKind declarationKind,
+            ParseOptions options, ExpressionBodyPreference preference,
+            out ArrowExpressionClauseSyntax arrowExpression,
+            out SyntaxToken semicolonToken)
+        {
+            if (!block.TryConvertToExpressionBody(
```

Comment:
>! [](start = 16, length = 1)

super-nit: consider inverting the `if` ;-)

Reply:
I'm good :)